src/keystrokehandler.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import EmitterMixin from './utils/emittermixin.js';
+import KeyObserver from './engine/view/observer/keyobserver.js';
+import { getCode, parseKeystroke } from './utils/keyboard.js';
+
+/**
+ * Keystroke handler. Its instance is available in {@link ckeditor5.Editor#keystrokes} so features
+ * can register their keystrokes.
+ *
+ * E.g. an undo feature would do this:
+ *
+ *		editor.keystrokes.set( 'ctrl + Z', 'undo' );
+ *		editor.keystrokes.set( 'ctrl + shift + Z', 'redo' );
+ *		editor.keystrokes.set( 'ctrl + Y', 'redo' );
+ *
+ * @memberOf ckeditor5
+ */
+export default class KeystrokeHandler {
+	/**
+	 * Creates an instance of the keystroke handler.
+	 *
+	 * @param {engine.treeView.TreeView} editingView
+	 */
+	constructor( editor ) {
+		/**
+		 * The editor instance.
+		 *
+		 * @readonly
+		 * @member {ckeditor5.Editor} ckeditor5.KeystrokeHandler#editor
+		 */
+		this.editor = editor;
+
+		/**
+		 * Observer used to listen to events for easier keystroke handler destruction.
+		 *
+		 * @private
+		 * @member {utils.Emitter} ckeditor5.KeystrokeHandler#_observer
+		 */
+		this._observer = Object.create( EmitterMixin );
```

Comment:
> What's wrong in _listeners.listenTo? If it can listen or stop listening it seems to be exactly a listener. Observer should observe, not listen.

The object is called "listeners". It does not contain nor represent multiple listeners. It's a single object which observes other instances. Plural names would be good for collections, but this is not one. So if anything, I'd go with `_subscriber` or `_observer`. `_observer` is good, because it's the name we've already been using. `_listener` would be nice because the method we have is called `listenTo()`, but I like it much less. That makes me think that this is another point for our Emitter refactoring issue.

I'll rename it to `_listener` (SINGULAR).